test/library/operand.go

```diff
@@ -0,0 +1,44 @@
+package library
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	osinv1 "github.com/openshift/api/osin/v1"
+)
+
+// GrabOAuthServerConfig grabs the oauth-server configuration from the
+// openshift-authenticat/v4-0-config-system-cliconfig configmap
+func GrabOAuthServerConfig(cmClient corev1client.ConfigMapsGetter) (*osinv1.OsinServerConfig, error) {
+	config, err := cmClient.ConfigMaps("openshift-authentication").Get(context.TODO(), "v4-0-config-system-cliconfig", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	rawConfig := config.Data["v4-0-config-system-cliconfig"]
+
+	configObj := &osinv1.OsinServerConfig{}
+	if err := json.NewDecoder(bytes.NewBuffer([]byte(rawConfig))).Decode(configObj); err != nil {
+		fmt.Printf("decoded object dump: %s", configObj)
+		return nil, fmt.Errorf("failed to decode object: %v", err)
+	}
+
+	return configObj, nil
+}
+
+// GetIDPByName returns a pointer to the identity provider of a name from the
+// config.
+// Do not modify the returned value, it would modify the config value as well
```

Comment:
Would it make sense to make a copy to avoid burdening the caller with this restriction?